pkg/domain/filter.go

```diff
@@ -0,0 +1,26 @@
+package domain
+
+import (
+	"fmt"
+)
+
+// ConfigurationItem contains the meaningful elements from the AWS Config Configuration Item
+// (https://docs.aws.amazon.com/config/latest/developerguide/config-item-table.html) that are
+// required for filtering.
+type ConfigurationItem struct {
+	ResourceType string `json:"resourceType"`
+}
+
+// ConfigFilter is the expected form of filters applied to AWS Config events.
+type ConfigFilter interface {
```

Comment:
The naming convention to use by default is https://golang.org/doc/effective_go.html#interface-names . Under that guide this might be better named as `ConfigFilterer` with a `FilterConfig` method. It's a nit pick though

Reply:
Good callout, I'll fix this naming convention.

Reply:
Fixed.